drivers/hud_apr/app/models/hud_apr/generators/shared/fy2021/base.rb

```diff
@@ -400,6 +401,39 @@ def needs_ce_assessments?
       scope
     end
 
+    private def pit_enrollment_info(enrollments)
+      pit_dates = [1, 4, 7, 10].map { |month| pit_date(month: month, before: @report.end_date) }
+      pit_dates.map do |pit_date|
+        enrollments_for_date = enrollments.select do |enrollment|
+          enrolled = case enrollment.project_type
+          when 3, 13 # PSH/RRH
+            enrollment.first_date_in_program <= pit_date &&
+              (enrollment.last_date_in_program.nil? || enrollment.last_date_in_program > pit_date) && # Exclude exit date
+              enrollment.move_in_date.present? && # Check that move in date is present and is before the PIT data
+              enrollment.move_in_date <= pit_date
+          when 1, 2, 8, 9, 10 # Other residential
+            enrollment.first_date_in_program <= pit_date &&
+              (enrollment.last_date_in_program.nil? || enrollment.last_date_in_program > pit_date) # Exclude exit date
+          else # Other project types (4, 6, 11)
+            enrollment.first_date_in_program <= pit_date &&
+              (enrollment.last_date_in_program.nil? || enrollment.last_date_in_program >= pit_date) # Include the exit date
+          end
+          next false unless enrolled
+
+          enrollment.project_tracking_method != 3 || enrollment.service_history_services.bed_nights.on_date(pit_date).exists?
```

Comment:
We need to guard this with computed_project_type == 1 in case there was an override from NbN ES to something else.